### PR TITLE
Fixed several broken links for developer setup

### PR DESCRIPTION
**Pull request by cinlloc**
- Fixed also bad ticket number

### CODE REVIEW

**Comment by neflyte** [comment]
@cinlloc Thanks for your contribution! Please update your branch. :)

**Review by neflyte — CHANGES_REQUESTED** [review]
Please update your branch to pick up the latest changes.

File: site/content/contribute/more-info/server/developer-setup/arch.md [hunk 1]
```diff
@@ -71,7 +71,7 @@
     {"AndroidLatestVersion":"","AndroidMinVersion":"","DesktopLatestVersion":"","DesktopMinVersion":"","IosLatestVersion":"","IosMinVersion":"","status":"OK"}
     ```
 
-    **Note:** Browsing directly to http://localhost:8065/ will display a `404 Not Found` until the web app is configured. See [Web App Developer Setup]({{< ref "/contribute/more-info/webapp/developer-setup" >}}) and [Mobile App Developer Setup]({{< ref "/contribute/more-info/mobile/developer-setup" >}}) for additional setup.
+    **Note:** Browsing directly to http://localhost:8065/ will display a `404 Not Found` until the web app is configured. See [Web App Developer Setup](https://developers.mattermost.com/contribute/more-info/webapp/developer-setup/) and [Mobile App Developer Setup](https://developers.mattermost.com/contribute/more-info/mobile/developer-setup/) for additional setup.
```
Comment (neflyte) [inline site/content/contribute/more-info/server/developer-setup/arch.md:74]: The links were correct in their existing form.
Why was this change made?

**Review by neflyte — CHANGES_REQUESTED** [review]

**Comment by neflyte** [comment]
I believe this issue is caused by a build pipeline problem. I am taking steps to have it addressed.

**Comment by cwarnermm** [comment]
@cinlloc - We're continuing to investigate existing link formatting issues present on pages of the developer documentation as we suspect it's a build issue rather than a code issue per source file. We'll keep this PR open in the event that our investigation reveals that source code changes will address the issue instead.

**Comment by neflyte** [comment]
The developer docs use relative rather than absolute links to other pages on the same site; we'd like to keep this usage consistent. Since the build issue is being investigated but has no estimate on a fix or workaround, we're closing this PR.

@cinlloc If you find links to other docs pages that are using an absolute URL, please let us know! :slightly_smiling_face:

**Comment by cinlloc** [comment]
@neflyte understood your point about relative links. However, there were several others little fixes in this PR that could be interesting to merge. I'm available to push them without the relative link stuff, I you feel so (cf. other opened comments). Keep me in touch ;-)

File: site/content/contribute/more-info/server/developer-setup/arch.md [hunk 2]
```diff
@@ -7,7 +7,7 @@
      newgrp docker
      ```
 
-    **Note:** [MM-9791](https://github.com/mattermost/mattermost-server/pull/10872) introduced using [docker-compose](https://docs.docker.com/compose/) to manage containers. To preserve your data on upgrade, execute the following steps.
+    **Note:** [MM-7971](https://github.com/mattermost/mattermost-server/pull/10872) introduced using [docker-compose](https://docs.docker.com/compose/) to manage containers. To preserve your data on upgrade, execute the following steps.
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/arch.md:10]: Here I fix a ticket number that was the wrong one

File: site/content/contribute/more-info/server/developer-setup/centos.md [hunk 3]
```diff
@@ -32,7 +32,7 @@
     sudo chmod +x /usr/local/bin/docker-compose
     ```
 
-3. Install Go (modify installation to latest Go version from https://golang.org/dl/):
+3. Install Go (modify installation to latest Go version from <https://golang.org/dl/>):
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/centos.md:35]: Here it fixes a broken link to golang download page (it was my initial issue, but it gave me the opportunity to check all the links in this page)

File: site/content/contribute/more-info/server/developer-setup/osx.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
-1. Install and configure Docker CE: https://docs.docker.com/docker-for-mac/.
+1. Install and configure Docker CE: https://docs.docker.com/desktop/install/mac-install/.
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/osx.md:1]: Here a up-to-date link to docker doc

File: site/content/contribute/more-info/server/developer-setup/osx.md [hunk 5]
```diff
@@ -42,7 +42,7 @@
     make run-server
     ```
 
-    **Note:** If you see lot of `nc: bad address 'elasticsearch'` messages, eventually exiting with the message `Service elasticsearch:9200 did not start within 300 seconds. Aborting...`,  increase the memory available for the Docker Engine by going to [Docker's preferences -> Advanced](https://docs.docker.com/docker-for-mac/#advanced).
+    **Note:** If you see lot of `nc: bad address 'elasticsearch'` messages, eventually exiting with the message `Service elasticsearch:9200 did not start within 300 seconds. Aborting...`,  increase the memory available for the Docker Engine by going to [Docker's preferences -> Advanced](https://docs.docker.com/desktop/settings/mac/#advanced).
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/osx.md:45]: Here again this link has to be updated

File: site/content/contribute/more-info/server/developer-setup/windows-wsl.md [hunk 6]
```diff
@@ -1,6 +1,6 @@
 This is an unofficial guide. Community testing, feedback, and improvements are welcome and greatly appreciated.
 
-1. Install the Windows Subsystem for Linux: https://docs.microsoft.com/en-us/windows/wsl/install-win10.
+1. Install the Windows Subsystem for Linux: https://learn.microsoft.com/en-us/windows/wsl/install.
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/windows-wsl.md:3]: This is the target of the redirection

File: site/content/contribute/more-info/server/developer-setup/windows-wsl.md [hunk 7]
```diff
@@ -18,12 +18,12 @@ This is an unofficial guide. Community testing, feedback, and improvements are w
 
 2. Install and set up Docker:
 
-    * Install Docker for Windows: https://docs.docker.com/docker-for-windows/.
+    * Install Docker for Windows: https://docs.docker.com/desktop/install/windows-install/.
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/windows-wsl.md:21]: New docker installation doc for windows (broken link)

File: site/content/contribute/more-info/server/developer-setup/windows-wsl.md [hunk 8]
```diff
@@ -44,7 +44,7 @@ This is an unofficial guide. Community testing, feedback, and improvements are w
     sudo chmod +x /usr/local/bin/docker-compose
     ```
 
-3. Install Go using bash (modify installation to latest Go version from https://golang.org/dl/):
+3. Install Go using bash (modify installation to latest Go version from <https://golang.org/dl/>):
```
Comment (cinlloc) [inline site/content/contribute/more-info/server/developer-setup/windows-wsl.md:47]: Here again the original golang broken link

**Comment by cwarnermm** [comment]
Thanks, @cinlloc! We really appreciate you surfacing links that would benefit from updates, and appreciate your understanding that we're aiming for consistent internal links overall. Very open to you updating your PR with these fixes!

File: site/content/contribute/more-info/server/developer-setup/arch.md (hunk 1 above)
Comment (neflyte) [inline site/content/contribute/more-info/server/developer-setup/arch.md:74]: Please revert these changes; we use relative links in the developer docs and we would like to keep them consistent.

**Review by neflyte — CHANGES_REQUESTED** [review]

File: site/content/contribute/more-info/server/developer-setup/arch.md (hunk 1 above)
Comment (cinlloc) [reply site/content/contribute/more-info/server/developer-setup/arch.md]: :ok_hand: Reverted all relative links changes in 2d88ef36fbb69e919fee7a9da850b6f2347c41db. Let me know if it's ok, I will do necessary fixup/rebase to clean-up the git tree of this PR.

**Review by cwarnermm — APPROVED** [review]
Thank you so much, @cinlloc!

File: site/content/contribute/more-info/server/developer-setup/arch.md (hunk 1 above)
Comment (cinlloc) [reply site/content/contribute/more-info/server/developer-setup/arch.md]: Rebase and fixup done.
